Update Bazel to 0.5.0 in kubekins-e2e and use Bazel's self-hosted JDK

ixdy opened the pull request:
As I somewhat suspected, it seems like bazel + jdk7 doesn't really work correctly, and jdk7 is going to stop being supported in 0.5.1 anyway.

Thankfully Bazel now ships with an installer with a self-hosted JDK for places where a JDK isn't already installed, like our `kubekins-e2e` container. As an added bonus, this image is also ~245MB smaller.

[Early test results](http://prow.k8s.io/log?pod=pull-kubernetes-e2e-gce-bazel-14) look promising...

(Also enabled privileged mode for `pull-kubernetes-e2e-gce-bazel`, since that's apparently needed for Bazel sandboxing.)

krzyzacy commented:
Is the e2e image change going to affect regular e2e tests? (We will be updating the tags pretty often recently), I'm asking since [gce-canary](https://k8s-testgrid.appspot.com/test-infra#gce-canary) has been pretty flaky yesterday... did you push a new kubekins-e2e image?

ixdy commented:
It would eventually affect regular e2e tests, though they don't use bazel, so the effect should be minimal.

I did push new images last night to test, which may explain gce-canary flakiness. I didn't notice you were also pushing new images at the time.

If we're happy with the changes in this PR, I can rebase for latest kubetest etc before merging.

krzyzacy commented:
/lgtm

ixdy commented:
Rebased and re-pushed images. Added `securityContext` and `TEST_TMPDIR` to the gce-bazel jobs, along with updating the kubekins-e2e-prow image used.

This only affects the gce-bazel jobs (for now), so this should be safe to merge. Doing so...
